### ISSUE TITLE
wxFindWindowAtPoint reports the wrong window page when inside a wxNotebook (Windows 10)

### ISSUE THREAD

**Issue by skeetor** [wxWidgets/wxWidgets#23589]
I created a wxNotebook with several pages in it and activated the first. Now I move the mouse over the page and do the following:

```
wxPoint mousePos = ::wxGetMousePosition();
wxWindow *window = wxFindWindowAtPoint(mousePos);
```

My expectation would be that `window` contains now a pointer to the window in the first (active) page, but `window` ALWAYS points to the last page instead.

I'm using the wxWindows library from github with the latest master, built with Visual Studio 2019 x64 Debug build.

**Comment by skeetor** [comment]
After looking into this a bit further, I noticed some strange behavior with the notebook itself, but I couldn't pin it down yet. I investigate this and report back when I found the problem.

**Comment by skeetor** [comment]
I identified the problem. So I'm create a new window and then insert it into the notebook. Afterwards I call `Show` on the window to make it visible.

```
  nb->InsertPage(i, userWindow, userWindow->GetName(), info.isActive());
  userWindow->Show();
```

when I revers this then everything works as expected:

```
  userWindow->Show();
  nb->InsertPage(i, userWindow, userWindow->GetName(), info.isActive());
```

Obviously I can easily fix this in my code, but IMO this should be automatically done by the notebook. When a page is added, it should make sure that Show is called, otherwise the behavior on Windows is strange. The notebook looks to the user as you would expected it, but when you call `wxFindWindowAtPoint` then Windows thinks the active page is the first one added. 

If you want, then I can add this to the `wxNotebook` and submit a pull request for it.

**Comment by vadz** [comment]
Sorry, I'm confused: if you pass `false` to `InsertPage()`, the page shouldn't be visible, normally, as it's hidden until it's actually shown. In fact, it's wrong to call `Show()` on it directly and we should probably detect this and trigger an assert failure or something -- the only way to show a page in a book control is by changing the current selection to it in the book control itself.

**Comment by skeetor** [comment]
I just tested what happens when I remove the call to `Show` and then it also seems to work. So this behavior only appears when I call `Show` after the insert.

**Comment by vadz** [comment]
> So it seems your suggestion to trigger an assert should definitely be implemented as this leads to an unexpected behavior which is not so easy to find.

The trouble is that it's difficult to implement such an assert. We can't really do it in `wxWindow::Show()` as it knows nothing (and shouldn't know anything) about `wxNotebook`, but `wxNotebook` itself can't intercept calls to `Show()` on its pages.

Perhaps we could check if the page we're about to show is already shown and assert if it is?

>  OTOH, When I add this window to a wxFrame instead, then I have to call Show, otherwise the window is not visible there.

This really shouldn't be the case, all non-toplevel windows are visible by default.

**Comment by skeetor** [comment]
> The trouble is that it's difficult to implement such an assert. We can't really do it in `wxWindow::Show()` as it knows nothing (and shouldn't know anything) about `wxNotebook`, but `wxNotebook` itself can't intercept calls to `Show()` on its pages.

Yeah. I was thinking on way how to detect  this, but I have no idea, as those calls are totally unrelated to each other and the notebook doesn't know what the user calls on his page window. Probably the only way would be if a page can only be a wxNotebookPage instead of a wxWindow. Then the call to Show can be disabled in that class. But this would be a rather drastic change and probably breaks a lot of existing code.

> Perhaps we could check if the page we're about to show is already shown and assert if it is?

That wouldn't help. As described above, it's the other way around. If the window is shown before it is inserted, then everything s fine. Only if the window is inserted and THEN it is shown it will cause a problem.
